metal_header/sifive_i2c0.c++

```diff
@@ -0,0 +1,312 @@
+                      std::to_string(pinmux_dest));
+
+      add_inline_body(pinmux_source_selector_func,
+                      "(uintptr_t)i2c == (uintptr_t)&__metal_dt_" + n.handle(),
+                      std::to_string(pinmux_source));
+
+      add_inline_body(interrupt_line_func,
+                      "(uintptr_t)i2c == (uintptr_t)&__metal_dt_" + n.handle(),
+                      std::to_string(irline));
+    }
+
+    count++;
+  });
+
+  os << "\n";
+  os << "/* --------------------- sifive_i2c0 ------------ */\n";
+
+  if (num_i2cs != 0) {
+    add_inline_body(control_base_func, "else", "0");
+    add_inline_body(control_size_func, "else", "0");
+    add_inline_body(clock_func, "else", "NULL");
+    add_inline_body(pinmux_func, "else", "NULL");
+    add_inline_body(pinmux_output_selector_func, "else", "0");
+    add_inline_body(pinmux_source_selector_func, "else", "0");
+    add_inline_body(interrupt_line_func, "else", "0");
+
+    emit_inline_def(control_base_func, "sifive_i2c0");
+    delete control_base_func;
+    emit_inline_def(control_size_func, "sifive_i2c0");
+    delete control_size_func;
+    emit_inline_def(clock_func, "sifive_i2c0");
+    delete clock_func;
+    emit_inline_def(pinmux_func, "sifive_i2c0");
+    delete pinmux_func;
+    emit_inline_def(pinmux_output_selector_func, "sifive_i2c0");
+    delete pinmux_output_selector_func;
+    emit_inline_def(pinmux_source_selector_func, "sifive_i2c0");
+    delete pinmux_source_selector_func;
+    emit_inline_def(num_interrupts_func, "sifive_i2c0");
+    delete num_interrupts_func;
+    emit_inline_def(interrupt_parent_func, "sifive_i2c0");
+    delete interrupt_parent_func;
+    emit_inline_def(interrupt_line_func, "sifive_i2c0");
+    delete interrupt_line_func;
+  }
+  os << "\n";
+}
+
+void sifive_i2c0::declare_structs() {
+  dtb.match(std::regex(compat_string),
+            [&](node n) { emit_struct_decl("sifive_i2c0", n); });
+}
+
+void sifive_i2c0::define_structs() {
+  dtb.match(std::regex(compat_string), [&](node n) {
+    emit_struct_begin("sifive_i2c0", n);
+    emit_struct_field("i2c.vtable", "&__metal_driver_vtable_sifive_i2c0.i2c");
+    emit_struct_end();
+  });
+}
+
+void sifive_i2c0::create_handles() {
+  emit_def("__METAL_DT_MAX_I2CS", std::to_string(num_i2cs));
+
+  emit_struct_pointer_begin("sifive_i2c0", "__metal_i2c_table", "[]");
+  if (num_i2cs) {
+    int i = 0;
+    dtb.match(std::regex(compat_string), [&](node n) {
+      os << "\t\t\t\t\t&__metal_dt_" << n.handle();
+      if ((i + 1) == num_i2cs) {
+        os << "};\n\n";
+      } else {
+        os << ",\n";
+      }
+      i++;
+    });
+  } else {
+    emit_struct_pointer_end("NULL");
+  }
+
+  if (num_i2cs) {
+    int i = 0;
+    dtb.match(std::regex(compat_string), [&](node n) {
+      emit_def_handle("__METAL_DT_SIFIVE_I2C0_HANDLE" + std::to_string(i), n,
+                      ".i2c");
+      i++;
+    });
+  }
```

Comment:
We should remove these handles, the right way to get access to I2C devices is with `metal_i2c_get_device()`.

Reply:
handles removed.